upload/catalog/controller/smailyforopencart/rss.php

```diff
@@ -86,7 +98,7 @@ public function index() {
     public function getCategoryIdByName($name) {
         $this->load->model('catalog/category');
         // Load all categories.
-        $categories = $this->model_catalog_category->getCategories();
+        $categories = $this->model_catalog_category->getCategories(array());
```

Comment:
Please take a look a the `getCategories` function definition in `catalog/model/catalog/category.php` and adjust the function usage here accordingly.

Reply:
I don't understand, what is wrong with it currently?

Reply:
If you look at the definition:

```php
public function getCategories($parent_id = 0) {
    // ...
}
```

you can see 2 things about the function's argument:
- it already has a default value;
- it requires the value to be an integer;

The problem is that you are providing a value of `array()` to `$parent_id`, which does get cast into `0` through `(int)$parent_id`, but is still wrong usage of the function.

The simplest solution to the problem is to not provide any arguments to the function call.